tracing-subscriber/src/fmt/mod.rs

```diff
@@ -1131,7 +1134,22 @@ pub fn try_init() -> Result<(), Box<dyn Error + Send + Sync + 'static>> {
     #[cfg(feature = "env-filter")]
     let builder = builder.with_env_filter(crate::EnvFilter::from_default_env());
 
-    builder.try_init()
+    let subscriber = builder.finish();
+    #[cfg(not(feature = "env-filter"))]
+        let targets = match std::env::var("RUST_LOG") {
+        Ok(var) => {
+            crate::filter::Targets::from_str(&var).map_err(|e|{eprintln!("Ignoring `RUST_LOG`: {:?}", e);}).unwrap_or_default()
+        }
+        Err(std::env::VarError::NotPresent) => {crate::filter::Targets::new()}
+        Err(e) => {
+            eprintln!("Ignoring `RUST_LOG`: {:?}", e);
+            crate::filter::Targets::new()
+        }
+    }.with_default(crate::fmt::Subscriber::DEFAULT_MAX_LEVEL);
```

Comment:
calling `with_default` on the result of the match will set the default max level in all cases, even if the filter parsed from the env var had a default level. for example, if the user runs a program with `RUST_LOG=debug,foo=trace`, where `Debug` is the default level, this will overwrite the `Debug` default with `Info`. instead, I think we should only overwrite the default max level when no filter was parsed from the env var:
```suggestion
        Err(std::env::VarError::NotPresent) => {
            crate::filter::Targets::new().with_default(crate::fmt::Subscriber::DEFAULT_MAX_LEVEL)
        }
        Err(e) => {
            eprintln!("Ignoring `RUST_LOG`: {:?}", e);
            crate::filter::Targets::new().with_default(crate::fmt::Subscriber::DEFAULT_MAX_LEVEL)
        }
    };
```